destination/destination_integration_test.go

```diff
@@ -0,0 +1,284 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	"cloud.google.com/go/pubsublite"
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/config"
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/models"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+	"google.golang.org/api/option"
+)
+
+const (
+	payload = "Hello, 世界"
+)
+
+func TestDestination_WriteSuccess(t *testing.T) {
+	var (
+		ctx, cancel = context.WithCancel(context.Background())
+		dest        = NewDestination()
+		cfg         = prepareConfig(t)
+		is          = is.New(t)
+	)
+
+	prepareTest(t, is, cfg)
+
+	err := dest.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = dest.Open(ctx)
+	is.NoErr(err)
+
+	records := []sdk.Record{
+		{
+			Payload: sdk.Change{After: sdk.RawData(payload)},
+		},
+	}
+
+	n := 0
+	n, err = dest.Write(ctx, records)
+	is.NoErr(err)
+	is.Equal(n, len(records))
+
+	cancel()
```

Comment:
Is it possible/practical to also get the message from Pub/Sub, and check if it was written correctly?

Reply:
Awesome :+1: